plugins/org.wso2.developerstudio.visualdatamapper.diagram/src/org/wso2/developerstudio/datamapper/diagram/edit/parts/DataMapperRootEditPart.java

```diff
@@ -16,65 +16,319 @@
+        ISchemaTransformer schemaTransformer = null;
+        try {
+            schemaTransformer = SchemaTransformerRegistry.getInstance().getSchemaTransformer().newInstance();
+        } catch (InstantiationException | IllegalAccessException e) {
+//          log.error(e);
+        }
+        // Model root of input schema tree
+        EList<TreeNode> TreeNodesList = null;
+        if (type.equals("input")){
+            TreeNodesList = ((DataMapperRoot) datamapperRoot).getInput().getTreeNode();
+        }
+        else {
+            TreeNodesList = ((DataMapperRoot) datamapperRoot).getOutput().getTreeNode();
+        }
+        File schemaFile = null;
+        String content = null;
+        
+        // If a tree node is found, continue saving
+        if (null != TreeNodesList && !TreeNodesList.isEmpty()) {
+            TreeNodeImpl inputTreeNode = (TreeNodeImpl) TreeNodesList.get(0);
+//            content = schemaTransformer.getSchemaContentFromModel(inputTreeNode, schemaFile);
+            content = schemaTransformer.getSchemaContentFromModel(inputTreeNode, schemaFile, true, content);  //.getSchemaContentFromModel(inputTreeNode, schemaFile);
+        }
+        // Empty tree node, clear the file
+        else {
+            content = "";
+        }
+        return content;
+    }
+    
+    private static void writeUsingFiles(String data) {
+        try {
+            Files.write(Paths.get("/Users/ayodhya/PycharmProjects/Data_mapping/files_java.json"), data.getBytes());
+        } catch (IOException e) {
+            e.printStackTrace();
+        }
+    }
+    
+    public List<String> getMappings(String inputSchema, String outputSchema) {
+        HttpURLConnection connection = null;  
 
+//        Gson gson = new Gson();
+//        String jsonIn = gson.toJson(inputSchema);
+//        String jsonOut = gson.toJson(outputSchema);
+        
+        // Create one set of schemas
+        JSONArray schemas = new JSONArray();
+        schemas.put(inputSchema);
+        schemas.put(outputSchema);
+        
+        String schemasToSend = schemas.toString();
+        
+        final HttpClient httpClient = new HttpClient();
+        
+        // Send file1
+        StringRequestEntity requestEntity;
+        String outString;
+        List<String> myList = null;
+        try {
+            requestEntity = new StringRequestEntity(schemasToSend,"application/json","UTF-8");
+            PostMethod postMethod = new PostMethod("http://127.0.0.1:5000/uploader");
+            postMethod.setRequestEntity(requestEntity);
+            try {
+                int statusCode = httpClient.executeMethod(postMethod);
+                outString = postMethod.getResponseBodyAsString();
+                myList = new ArrayList<String>(Arrays.asList(outString.split("#")));
+            } catch (HttpException e) {
+                // TODO Auto-generated catch block
+                e.printStackTrace();
```

Comment:
Shall we log the error instead of printing the stack trace?

Reply:
Done. Please check